src/target/riscv/riscv.c

```diff
@@ -2741,6 +2742,12 @@ static int riscv_run_algorithm(struct target *target, int num_mem_params,
 		return ERROR_FAIL;
 	}
 
+	// Restore Interrupts
+	LOG_DEBUG("Restoring Interrupts");
+	buf_set_u64(mstatus_bytes, 0, info->xlen, current_mstatus);
+	register_set(&target->reg_cache->reg_list[REG_MSTATUS], mstatus_bytes);
+	info->mstatus_actual = current_mstatus;
```

Comment:
I don't think you should. register_set/register_get are the entry points when a user from gdb wants to change registers. mstatus_actual is an internal detail that's not exposed to the outside world, which this code is part of.